client/transaction-pool/src/revalidation.rs

```diff
@@ -38,6 +38,8 @@ const MIN_BACKGROUND_REVALIDATION_BATCH_SIZE: usize = 20;
 
 /// Payload from queue to worker.
 struct WorkerPayload<Api: ChainApi> {
+	/// The last known best_block.
+	best_block: Option<BlockId<Api::Block>>,
```

Comment:
I feel like this isn't required. `at` should always be the best block. If there is a race condition between running revalidation task and import new blocks we should extend `ChainApi` to simply return current best block and always validate against this.